hud_api_replace/geocode.py

```diff
@@ -25,18 +34,19 @@ def _extract_zip_code(key):
 
 def geocode_get_data(address, zip_only=False):
     """Main function to obtain geocoding information."""
-    address = str(address) + ", usa"
     try:
         cached = CachedGeodata.objects.all().filter(expires__gte=time.time()).get(key=address)
     except ObjectDoesNotExist:
         if zip_only:
-            mb_result = Mapbox(address ,types='postcode')
+            mb_result = PermanentMapbox(address,
+                                        types='postcode',
+                                        country='us')
```

Comment:
This logic appears to work well except the fallback to `geocoder.osm` then breaks the results.

Take for example these cases that return incorrect results:

```py
>>> geocode_get_data('00000', zip_only=True)      
{'zip': {'lat': 41.9451569974435, 'lng': 20.6161373631612, 'zipcode': '00000'}}
>>> geocode_get_data('00001', zip_only=True)
{'zip': {'lat': 51.149086464458, 'lng': 71.4951932054471, 'zipcode': '00001'}}
```

Both of these are invalid zip codes and the results seem random. If you omit `zip_only`, on the other hand, behavior seems correct:

```py
>>> geocode_get_data('00000')
{'zip': {'lat': None, 'lng': None, 'zipcode': '00000'}}
>>> geocode_get_data('18020')
{'zip': {'lat': 40.668484, 'lng': -75.32711, 'zipcode': '18020'}}
```

The reason is that with `zip_only` Mapbox properly returns no results:

```py
>>> PermanentMapbox('00000', types='postcode', country='us')
<[ERROR - No results found] Mapbox - Geocode>
>>> PermanentMapbox('18020', types='postcode', country='us')
<[OK] Mapbox - Geocode [18020, Pennsylvania, United States]>
```

But then the `geocode.osm` fallback just looks it up and returns some random result:

```py
>>> osm('00000').latlng
[41.9451569974435, 20.6161373631612]
```

Can we just remove the OSM fallback and store null values for invalid results? It might require some downstream changes in cfgov-refresh to properly handle but it'd be worth it and make the API a bit more useful.